mifospay/src/main/java/org/mifos/mobilewallet/mifospay/base/BaseActivity.kt

```diff
@@ -0,0 +1,177 @@
+package org.mifos.mobilewallet.mifospay.base
+
+import android.app.ProgressDialog
+import android.content.pm.ActivityInfo
+import android.support.v4.app.Fragment
+import android.support.v4.app.FragmentManager
+import android.support.v4.widget.SwipeRefreshLayout
+import android.support.v7.widget.Toolbar
+import android.view.MenuItem
+import com.mifos.mobile.passcode.BasePassCodeActivity
+import org.mifos.mobilewallet.mifospay.MifosPayApp
+import org.mifos.mobilewallet.mifospay.R
+import org.mifos.mobilewallet.mifospay.injection.component.ActivityComponent
+import org.mifos.mobilewallet.mifospay.injection.component.DaggerActivityComponent
+import org.mifos.mobilewallet.mifospay.injection.module.ActivityModule
+import org.mifos.mobilewallet.mifospay.passcode.ui.PassCodeActivity
+
+/**
+ * Created by naman on 16/6/17.
+ */
+open class BaseActivity : BasePassCodeActivity(), BaseActivityCallback {
+    var toolbar: Toolbar? = null
+    override var swipeRefreshLayout: SwipeRefreshLayout? = null
+    var progressDialog: ProgressDialog? = null
+    var activityComponent: ActivityComponent? = null
+        get() {
+            if (field == null) {
```

Comment:
In Kotlin you can simply leverage the let operator as: `?.let{ // code here }`. 
Also if you have used Android Studio's built-in Java to Kotlin converter please re-check the changes done as the tool just does the bare bones work and doesn't actually make full use of Kotlin's features.